mlir/lib/Transforms/InlinerPass.cpp

```diff
@@ -100,9 +100,12 @@ void InlinerPass::runOnOperation() {
     return signalPassFailure();
   }
 
+  // By default, assume that any inlining is profitable.
+  auto profitabilityCb = [](Inliner::ResolvedCall &) { return true; };
```

Comment:
Can we add a pass option `int maxOperationCountToInline`  and when it is set walk the callee and online inline when below the threshold?

That would allow to write a test and showcase the feature here.

We could also do it as a percentage and count the number of operations in the calller, the callee, and only inline if the callee size if within x% of the caller size.

Reply:
Sure, I can add it to showcase the usage. The percentage threshold sounds better to me as it involves analyzing both the caller and the callee.  One thing I am not clear about is what to consider "a caller".  I think it should be the first parent that is `IsolatedFromAbove`, but I am open to other ideas.

Reply:
This seems reasonable for the generic inliner pass I guess.